### PR TITLE
Add air functionality

### PR DESCRIPTION
**Pull request by har777**
Adds the following flags to the vm:
1. `--air_public_input`
2. `--air_private_input`

This PR adds everything except the `memory_segments` and `public_memory` keys for the `--air_public_input` file

### CODE REVIEW

File: pkg/runners/zero/air_input.go [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+							continue
+						}
+						valueBig := big.Int{}
+						value.Felt.BigInt(&valueBig)
+						valueHex := fmt.Sprintf("0x%x", &valueBig)
+						builtinValues = append(builtinValues, AirPrivateBuiltinRangeCheck{Index: index, Value: valueHex})
+					}
+					airPrivateInput.RangeCheck = builtinValues
+				}
+			case "bitwise":
```
Comment (har777) [inline pkg/runners/zero/air_input.go:73]: We could also move this functionality to the respective builtin files

File: pkg/runners/zero/air_input.go [hunk 2]
```diff
@@ -0,0 +1,416 @@
...
+	return AirPublicInput{
+		Layout:        runner.layout.Name,
+		RcMin:         rcMin,
+		RcMax:         rcMax,
+		NSteps:        len(runner.vm.Trace),
+		DynamicParams: nil,
+		// TODO: yet to be implemented
+		MemorySegments: make(map[string]AirMemorySegmentEntry),
+		// TODO: yet to be implemented
+		PublicMemory: make([]AirPublicMemoryEntry, 0),
```
[the remark this answers is not in the document]
Comment (har777) [reply pkg/runners/zero/air_input.go:18]: Couldn't figure out what the rust vm was doing

File: cmd/cli/main.go [hunk 3]
```diff
@@ -179,6 +195,46 @@ func main() {
...
+								return err
+							}
+							airPublicInputJson, err := json.MarshalIndent(airPublicInput, "", "  ")
+							if err != nil {
+								return err
+							}
+							err = os.WriteFile(airPublicInputLocation, airPublicInputJson, 0644)
+							if err != nil {
+								return fmt.Errorf("cannot write air_public_input: %w", err)
+							}
```
[the remark this answers is not in the document]
Comment (har777) [reply cmd/cli/main.go:211]: Only twice. Doesn't warrant a utility method imo.

File: pkg/runners/zero/air_input.go (hunk 1 above)
Comment (har777) [reply pkg/runners/zero/air_input.go]: Done

File: pkg/runners/zero/air_input.go [hunk 4]
```diff
@@ -0,0 +1,416 @@
...
+type AirPrivateInput struct {
+	TracePath  string                        `json:"trace_path"`
+	MemoryPath string                        `json:"memory_path"`
+	Pedersen   []AirPrivateBuiltinPedersen   `json:"pedersen"`
+	RangeCheck []AirPrivateBuiltinRangeCheck `json:"range_check"`
+	Ecdsa      []AirPrivateBuiltinECDSA      `json:"ecdsa"`
+	Bitwise    []AirPrivateBuiltinBitwise    `json:"bitwise"`
+	EcOp       []AirPrivateBuiltinEcOp       `json:"ec_op"`
+	Keccak     []AirPrivateBuiltinKeccak     `json:"keccak"`
+	Poseidon   []AirPrivateBuiltinPoseidon   `json:"poseidon"`
```
[the remark this answers is not in the document]
Comment (har777) [reply pkg/runners/zero/air_input.go]: Yes the rust VM does add keys for `add_mod` and `mul_mod` in its latest build with feature `mod_builtin` enabled. We need to first implement the builtin completely.

File: cmd/cli/main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (cicr99) [reply cmd/cli/main.go:211]: This seemed like a general logic, not only for air inputs. Turns out we do have this utility implemented already, but for ZeroProgram structs only `MarshalToFile` in `pkg/parsers/zero/zero.go`.
We can leave it like this, and do some refactoring after.

**Review by cicr99 — APPROVED** [review]

File: pkg/runners/zero/air_input.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (cicr99) [reply pkg/runners/zero/air_input.go]: Ok, I've created a separate issue to look into that #661

**Pull request merged**